File: src/main/java/com/tribalscale/test/mapper/AccountMapper.java
package com.tribalscale.test.mapper;

import com.tribalscale.test.domain.Account;
import com.tribalscale.test.domain.payload.AccountPayload;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface AccountMapper {
    Account payloadToEntity(AccountPayload payload);

    AccountPayload entityToPayload(Account entity);
}

File: src/main/java/com/tribalscale/test/mapper/AccountMapper.kt
package com.tribalscale.test.mapper

import com.tribalscale.test.domain.Account
import com.tribalscale.test.domain.payload.AccountPayload
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface AccountMapper {
    fun payloadToEntity(payload: AccountPayload): Account
    fun entityToPayload(entity: Account): AccountPayload
}